libs/config/src/analyze/lints/collect_stringtables.rs

```diff
@@ -0,0 +1,73 @@
+use hemtt_common::config::{LintConfig, ProjectConfig};
+use hemtt_workspace::{
+    lint::{AnyLintRunner, Lint, LintRunner}, reporting::{Codes, Processed}
+};
+
+use crate::{analyze::LintData, Value};
+
+crate::analyze::lint!(LintColectStringtables);
+
+impl Lint<LintData> for LintColectStringtables {
+    fn display(&self) -> bool {
+        false
+    }
+
+    fn ident(&self) -> &'static str {
+        "collect_stringtable"
+    }
+
+    fn sort(&self) -> u32 {
+        0
+    }
+
+    fn description(&self) -> &'static str {
+        "collect_stringtable"
+    }
+
+    fn documentation(&self) -> &'static str {
+        r"This should not be visable?"
+    }
+
+    fn default_config(&self) -> LintConfig {
+        LintConfig::warning()
+    }
+
+    fn runners(&self) -> Vec<Box<dyn AnyLintRunner<LintData>>> {
+        vec![Box::new(Runner)]
+    }
+}
+
+struct Runner;
+
+impl LintRunner<LintData> for Runner {
+    type Target = Value;
+    fn run(
+        &self,
+        _project: Option<&ProjectConfig>,
+        _config: &LintConfig,
+        processed: Option<&Processed>,
+        target: &Value,
+        data: &LintData,
+    ) -> Codes {
+        let Some(processed) = processed else {
+            return vec![];
+        };
+        let Value::Str(cstring_data) = target else {
+            return vec![];
+        };
+        let cstring_value = cstring_data.value();
+
+        if cstring_value.to_lowercase().starts_with("str_") || cstring_value.to_lowercase().starts_with("$str_") {
```

Comment:
more self notes

```
x1 = "str_ace_common_cancel";  // "str_ace_common_cancel"
x2 = "$str_ace_common_cancel"; // "$str_ace_common_cancel"
x3 = "$STR_ace_common_cancel"; // "Cancel"
x4 = "$STR_ace_coMMon_cancel"; // "Cancel"

localize "str_aCe_common_canCel" // "Cancel"
localize "$str_aCe_common_canCel" // "Cancel"
```

config needs cased "$STR" to start but then doesn't care about case
sqf doesn't care about case or if starting with a $
this might be newish - otherwise I don't know why we have CSTRING and LSTRING macros?

Reply:
After the wiki $ in localize was introduced in 2.04